modules/decision/search_pattern.py

```diff
@@ -0,0 +1,242 @@
+            side_length -= (self.__search_gap_depth - self.__search_gap_width) / 2
+
+        # If we are on the last side of the square, we will continue until the next square to spiral
+        if self.__current_side_in_square == 3:
+            side_length += self.__search_gap_width
+
+        # Calculate the number of stops needed along the current side
+        self.__max_pos_on_side = ceil(abs(side_length) / self.__search_gap_depth)
+
+        self.__travel_gap = side_length / self.__max_pos_on_side
+
+    def set_target_location(self):
+        """
+        Calculate and set the next target location for the drone to move to.
+        Returns true if the new target location is a spot we should scan at. If false, it means
+        this point is an intermediate point so that the next one is facing the correct direction
+        """
+
+        # If we've reached the end of the current side, move to the next side
+        if self.__current_pos_on_side >= self.__max_pos_on_side:
+            self.__current_pos_on_side = -1  # Reset position counter for the new side
+            self.__current_side_in_square = (
+                self.__current_side_in_square + 1
+            ) % 4  # Move to the next side
+
+            if self.__current_side_in_square == 0:  # If completed this square
+                self.__current_square += 1
+                self.__calculate_square_corners()
+
+            self.__calculate_side_of_square()
+
+        # For the first position on a side, we set the drone a small amount off of the position so
+        # that the next move turns the drone to face the correct direction
+        if self.__current_pos_on_side == -1:
+            if self.__current_side_in_square == 0:
+                self.__target_posx = self.__current_corner[0] - self.__small_adjustment
+                self.__target_posy = self.__current_corner[1]
+            elif self.__current_side_in_square == 1:
+                self.__target_posx = self.__current_corner[0]
+                self.__target_posy = self.__current_corner[1] + self.__small_adjustment
+            elif self.__current_side_in_square == 2:
+                self.__target_posx = self.__current_corner[0] + self.__small_adjustment
+                self.__target_posy = self.__current_corner[1]
+            elif self.__current_side_in_square == 3:
+                self.__target_posx = self.__current_corner[0]
+                self.__target_posy = self.__current_corner[1] - self.__small_adjustment
+
+            # Increment the position counter
+            self.__current_pos_on_side += 1
+            return False
+
+        # Calculate the next target position based on the current fraction of the side covered
+        dist_to_move = self.__travel_gap * self.__current_pos_on_side
+        if self.__moving_horizontally:
+            self.__target_posx = self.__current_corner[0] + dist_to_move
+        else:
+            self.__target_posy = self.__current_corner[1] + dist_to_move
```

Comment:
I would prefer having using x += self.__travel_gap  and y += self.__travel_gap since these should represent the current state of the drone to avoid the multiplications and reassigning x and y